geps/gep-1767/index.md

```diff
@@ -0,0 +1,701 @@
+    //   Access-Control-Allow-Credentials: true
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:default=false
+    AllowCredentials bool `json:"allowCredentials,omitempty"`
+
+    // AllowMethods indicates which HTTP methods are supported 
+    // for accessing the requested resource.
+    // The method is case-sensitive.
+    //
+    // Config:
+    //   allowMethods: ["GET, PUT, POST, DELETE, PATCH, OPTIONS"]
+    // 
+    // Output:
+    //   Access-Control-Allow-Methods: GET, PUT, POST, DELETE, PATCH, OPTIONS
+    //
+    // A wildcard indicates that the requests with all HTTP methods are allowed.
+    // When responding to a credentialed requests, the gateway must specify 
+    // one or more HTTP methods in the value of the Access-Control-Allow-Methods response header, 
+    // instead of specifying the * wildcard.
+    //
+    // Config:
+    //   allowMethods: ["*"]
+    // 
+    // Output:
+    //   Access-Control-Allow-Methods: ["*"]
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +listType=set
+    // +kubebuilder:validation:MaxItems=16
+    AllowMethods []string `json:"allowMethods,omitempty"`
+ 
+    // AllowHeaders indicates which HTTP request headers are supported 
+    // for accessing the requested resource.
+    // Header names are not case sensitive.
+    //
+    // Config:
+    //   allowHeaders: ["DNT", "Keep-Alive", "User-Agent", "X-Requested-With", "If-Modified-Since", "Cache-Control", "Content-Type", "Range", "Authorization"]
+    // Output:
+    //   Access-Control-Allow-Headers: DNT, Keep-Alive, User-Agent, X-Requested-With, If-Modified-Since, Cache-Control, Content-Type, Range, Authorization
+    //
+    // A wildcard indicates that the requests with all HTTP headers are allowed.
+    // When responding to a credentialed requests, the gateway must specify 
+    // one or more HTTP headers in the value of the Access-Control-Allow-Headers response header, 
+    // instead of specifying the * wildcard.
+    //
+    // Config:
+    //   allowHeaders: ["*"]
+    // Output:
+    //   Access-Control-Allow-Headers: *
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +listType=set
+    // +kubebuilder:validation:MaxItems=64
+    AllowHeaders []string `json:"allowHeaders,omitempty"`
+
+    // ExposeHeaders indicates which HTTP response headers are exposed to clients 
+    // for the cross-origin request that is not a "preflight" request.
+    // Header names are not case sensitive.
+    //
+    // Config:
+    //   exposeHeaders: ["Content-Security-Policy"]
+    // Output:
+    //   Access-Control-Expose-Headers: Content-Security-Policy
+    //
+    // A wildcard indicates that the responses with all HTTP headers are exposed to clients.
+    // When responding to a credentialed requests, the gateway must specify 
+    // one or more HTTP headers in the value of the Access-Control-Expose-Headers response header, 
+    // instead of specifying the * wildcard.
```

Comment:
Does "exposed to clients" mean "returned as headers in requests"? Seems like it, but if that's the case, that should be said specifically here. Once we define that, then we can use "exposed to clients" and "returned as headers" interchangeably.